src/bootstrap/src/core/sanity.rs

```diff
@@ -177,6 +177,30 @@ than building it.
             continue;
         }
 
+        // Check if there exists a built-in target in the list of supported targets.
+        let mut has_target = false;
+        let target_str = target.to_string();
+
+        let supported_target_list = Command::new(&build.config.initial_rustc)
+            .args(["--print", "target-list"])
+            .output()
+            .expect("Should have been able to fetch the target list.");
+
+        if let Some(stdout) = String::from_utf8(supported_target_list.stdout).ok() {
```

Comment:
There is a helper function for capturing cmd outputs: https://github.com/rust-lang/rust/blob/5557f8c9d08d7f3f680943dcf97b6d4a7eb13aea/src/bootstrap/src/utils/helpers.rs#L259